airflow/utils/parse.py

```diff
@@ -0,0 +1,168 @@
+                )
+            )
+        contents_dict[key].append(value)
+    return contents_dict, errors
+
+
+def _parse_yaml_file(file_path: str) -> Tuple[Dict[str, List[str]], List[FileSyntaxError]]:
+    """
+    Parse a file in the YAML format.
+
+    :param file_path: The location of the file that will be processed.
+    :type file_path: str
+    :return: Tuple with mapping of key and list of values and list of syntax errors
+    """
+    with open(file_path) as f:
+        content = f.read()
+
+    if not content:
+        return {}, [FileSyntaxError(line_no=1, message="The file is empty.")]
+    try:
+        contents_dict = yaml.safe_load(content)
+
+    except yaml.MarkedYAMLError as e:
+        return {}, [FileSyntaxError(line_no=e.problem_mark.line, message=str(e))]
+    if not isinstance(contents_dict, dict):
+        return {}, [FileSyntaxError(line_no=1, message="The file should contain the object.")]
+
+    return contents_dict, []
+
+
+def _parse_json_file(file_path: str) -> Tuple[Dict[str, Any], List[FileSyntaxError]]:
+    """
+    Parse a file in the JSON format.
+
+    :param file_path: The location of the file that will be processed.
+    :type file_path: str
+    :return: Tuple with mapping of key and list of values and list of syntax errors
+    """
+    with open(file_path) as f:
+        content = f.read()
+
+    if not content:
+        return {}, [FileSyntaxError(line_no=1, message="The file is empty.")]
+    try:
+        contents_dict = json.loads(content)
+    except JSONDecodeError as e:
+        return {}, [FileSyntaxError(line_no=int(e.lineno), message=e.msg)]
+    if not isinstance(contents_dict, dict):
+        return {}, [FileSyntaxError(line_no=1, message="The file should contain the object.")]
+
+    return contents_dict, []
+
+
+FILE_PARSERS = {
+    "env": _parse_env_file,
+    "json": _parse_json_file,
+    "yaml": _parse_yaml_file,
+}
+
+
+def _parse_file(file_path: str) -> Dict[str, Any]:
+    """
+    Based on the file extension format, selects a parser, and parses the file.
+
+    :param file_path: The location of the file that will be processed.
+    :type file_path: str
+    :return: Map of key (e.g. connection ID) and value.
+    """
+    if not os.path.exists(file_path):
+        raise AirflowException(f"File {file_path} was not found.")
+
+    log.debug("Parsing file: %s", file_path)
+
+    ext = file_path.rsplit(".", 2)[-1].lower()
+
+    if ext not in FILE_PARSERS:
+        raise AirflowException(
+            "Unsupported file format. The file must have the extension .env or .json or .yaml"
+        )
+
+    contents_dict, parse_errors = FILE_PARSERS[ext](file_path)
+
+    log.debug(
+        "Parsed file: len(parse_errors)=%d, len(contents_dict)=%d", len(parse_errors), len(contents_dict)
+    )
+
+    if parse_errors:
+        raise AirflowFileParseException(
+            "Failed to load the file.", file_path=file_path, parse_errors=parse_errors
+        )
```

Comment:
Just make the format specific parsers throw the errors directly?

Reply:
No problem for me to do this, but heads up that this is not new code, just cut from `secrets/local_filesystem.py`.